internal/pkg/callbacks/rolling_upgrade.go

```diff
@@ -0,0 +1,108 @@
+package callbacks
+
+import (
+	"github.com/sirupsen/logrus"
+	"github.com/stakater/Reloader/internal/pkg/util"
+	apps_v1beta1 "k8s.io/api/apps/v1beta1"
+	"k8s.io/api/core/v1"
+	"k8s.io/api/extensions/v1beta1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+//ItemsFunc is a generic function to return a specific resource array in given namespace
+type ItemsFunc func(kubernetes.Interface, string) []interface{}
+
+//ContainersFunc is a generic func to return containers
+type ContainersFunc func(interface{}) []v1.Container
+
+//UpdateFunc performs the resource update
+type UpdateFunc func(kubernetes.Interface, string, interface{}) error
+
+type ResourceTypeFunc func() string
+
+//RollingUpgradeFuncs contains generic functions to perform rolling upgrade
+type RollingUpgradeFuncs struct {
+	ItemsFunc        ItemsFunc
+	ContainersFunc   ContainersFunc
+	UpdateFunc       UpdateFunc
+	ResourceTypeFunc ResourceTypeFunc
```

Comment:
This is not needed. Callback functions are only used where you're not sure what the logic for doing a specific task is going to be. Here you know that the resource type is a literal string and can be passed as `deployment` etc